server/pipeline/topic.go

```diff
@@ -33,6 +33,7 @@ func publishToTopic(c context.Context, pipeline *model.Pipeline, repo *model.Rep
 		},
 	}
 	pipelineCopy := *pipeline
+	pipelineCopy.Workflows = nil // never send workflow lists
```

Comment:
From testing I think this is needed. If we need pipelines to be approved we do not get the workflow list after accepting by the update.

Reply:
AFAIK this function is sending an event to the UI with pipeline updates. If we approve a pipeline the workflows get parsed and were previously send via this event as part of the pipeline. Now we do not get the workflow list anymore until we refresh the full page doing a fetch for the pipeline containing the added workflows list.

```suggestion
```